training/heterogeneous-clusters/pt.grpc.sagemaker/hetero-pytorch-mnist.ipynb

```diff
@@ -0,0 +1,509 @@
+    "\n",
+    "At a high level, the notebook covers:\n",
+    "- Setting up SageMaker Studio Notebook \n",
+    "- Setting up the Training environment \n",
+    "- Submit a Training job\n",
+    "- Monitor and visualize the CloudWatch metrics\n",
+    "- Comparing time-to-train and cost-to-train\n",
+    "- Conclusion \n",
+    "---\n"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "fd1e5aca",
+   "metadata": {},
+   "source": [
+    "### A. Setting up SageMaker Studio notebook\n",
+    "\n",
+    "#### Step 1 - Upgrade SageMaker SDK and dependent packages \n",
+    "Heterogeneous Clusters for Amazon SageMaker model training was [announced](https://aws.amazon.com/about-aws/whats-new/2022/07/announcing-heterogeneous-clusters-amazon-sagemaker-model-training) on 07/08/2022. As a first step, ensure you have updated SageMaker SDK, PyTorch, and Boto3 client that enables this feature."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 51,
+   "id": "54ff1687",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "%%bash\n",
+    "python3 -m pip install --upgrade boto3 botocore awscli sagemaker"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "0d20b2f3",
+   "metadata": {},
+   "source": [
+    "#### Step 2 - Restart the notebook kernel \n",
+    "From the Jupyter Lab menu bar **Kernel > Restart Kernel...**"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "a9592cda",
+   "metadata": {},
+   "source": [
+    "#### Step 3 - Validate SageMaker Python SDK and PyTorch versions\n",
+    "Ensure the output of the cell below reflects:\n",
+    "\n",
+    "- SageMaker Python SDK version 2.98.0 or above, \n",
+    "- boto3 1.24 or above \n",
+    "- botocore 1.27 or above \n",
+    "- PyTorch 1.10 or above "
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 52,
+   "id": "0b0e3202",
+   "metadata": {},
+   "outputs": [
+    {
+     "name": "stdout",
+     "output_type": "stream",
+     "text": [
+      "Name: sagemaker\n",
+      "Version: 2.109.0\n",
+      "---\n",
+      "Name: torch\n",
+      "Version: 1.10.2+cpu\n",
+      "---\n",
+      "Name: boto3\n",
+      "Version: 1.24.72\n",
+      "---\n",
+      "Name: botocore\n",
+      "Version: 1.27.72\n"
+     ]
+    }
+   ],
+   "source": [
+    "!pip show sagemaker torch boto3 botocore |egrep 'Name|Version|---'"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "9176d868",
+   "metadata": {},
+   "source": [
+    "--------------\n",
+    "### B. Setting up the Training environment\n",
+    "\n",
+    "#### Step 1 - Import SageMaker components and set up the IAM role and Amazon S3 bucket"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 53,
+   "id": "594fce53",
+   "metadata": {},
+   "outputs": [
+    {
+     "name": "stdout",
+     "output_type": "stream",
+     "text": [
+      "arn:aws:iam::776941257690:role/service-role/AmazonSageMakerServiceCatalogProductsUseRole\n",
```

Comment:
Think you should clear your outputs. If you need to keep some outputs we can look into Papermill or something to hide some things and display others, but keep functionality for testing.